Sources/PackageGraph/PackageGraph.swift

```diff
@@ -33,8 +33,12 @@ public struct PackageGraph {
         return rootPackages.flatMap{$0.modules}.contains(module)
     }
 
+    /// The errors encountered while loading the package graph.
+    public let errors: [Swift.Error]
```

Comment:
Great.

To elaborate on the error reporting:  what I meant is that each function that can report issues is given an error-reporter that conforms to some protocol; this protocol can then later be extended to support warnings, fix-its, etc.  The Workspace can implement a reporter that records the errors that are encountered in the list (as in the PR) and also tells its own delegate (which can then choose to print it etc).